### DISCUSSION TITLE
Update graph with new incoming data and leave it unfilled

### DISCUSSION THREAD

**Discussion by ivevasiljevic** [patrykandpatrick/vico#600] [Questions] [answered]
Is it possible to feed the producer with a full x axis of values, for example from 0 to 300, and then in y axis only have values for 0 to 100. Leaving the rest of the graph unfilled, and then later when I have the next value just update the graph.

Something like this will explain it better:

![Screenshot 2024-03-06 at 12 55 04](https://github.com/patrykandpatrick/vico/assets/30770600/32812599-4afd-419f-9eb4-5ebefea6dd9f)

I've accomplished this, with having two lineSeries. One having the full width with -1 y values, and the other having width of the data it has. And then two rememberLineCartesianLayer setup as well so I can hide the first lineSeries with transparent color.

Is it possible to do it, with just one lineSeries and one layer?

Edit: Since I have a some kind of solution it is fine and all, but when I want to place a marker on the graph, the marker follows the placeholder line as well, I want it just to follow the line with real data.

**Comment by patrickmichalik** [comment]
Hello! Yup, you can do this with one series and one `LineCartesianLayer`. Rather than adding a placeholder series, please try adding the following to the `rememberLineCartesianLayer` call:

```kt
axisValueOverrider = remember { AxisValueOverrider.fixed(maxX = 300f) }
```

Note that a different approach should be used if the maximum _x_ value is to change with the data. If this is required, please let me know what kind of dependency is needed, and I’ll describe the appropriate approach.

**Reply by ivevasiljevic** [reply]
I actually figured it out, just as I wrote this question. It is funny how things workout just as you ask a question, some kind of magic😁

I implemented my own AxisValueOverrider, where I specify max and min values for x and y, based on the data that is being fed into the graph. I send this data using ExtraStore, so that when I update the data, x and y values update as well. Is this the right approach?

```
fun customAxisValueOverrider(axesExtraStoreKey: ExtraStore.Key<ChartData.AxesData>): AxisValueOverrider =
    object : AxisValueOverrider {
        override fun getMinY(
            minY: Float,
            maxY: Float,
            extraStore: ExtraStore,
        ) = floor(extraStore[axesExtraStoreKey].minY)

        override fun getMaxY(
            minY: Float,
            maxY: Float,
            extraStore: ExtraStore,
        ) = ceil(extraStore[axesExtraStoreKey].maxY)

        override fun getMaxX(
            minX: Float,
            maxX: Float,
            extraStore: ExtraStore
        ) = extraStore[axesExtraStoreKey].maxX

        override fun getMinX(
            minX: Float,
            maxX: Float,
            extraStore: ExtraStore
        ) = extraStore[axesExtraStoreKey].minX
    }
```

**Reply by patrickmichalik** [reply, marked as answer]
I’m glad you’ve found a solution, @ivevasiljevic! Yup, the answer to a question often becomes apparent right after you ask the question.

This looks good to me. It should be noted that if the final _x_ and _y_ ranges depend only on the data’s _x_ and _y_ ranges, you can avoid using extras and instead calculate the final minimum and maximum _x_ and _y_ values in the `AxisValueOverrider` functions based on the `minX`, `maxX`, `minY`, and `maxY` arguments. These represent the data’s minimum and maximum _x_ and _y_ values. If the final ranges depend on something more, however, extras should be used. (And even if they don’t, using extras is perfectly fine—it might just be a bit more complex.)
